sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py

```diff
@@ -2666,3 +2666,54 @@ def test_dsl_pipeline_with_data_transfer_copy_2urifolder(self, client: MLClient)
         }
         assert expected_job == actual_job
 
+
+    @pytest.mark.disable_mock_code_hash
+    def test_register_output_sdk_succeed(self, client: MLClient):
+        component = load_component(source="./tests/test_configs/components/helloworld_component.yml")
+        component_input = Input(type='uri_file', path='https://dprepdata.blob.core.windows.net/demo/Titanic.csv')
+
+        @dsl.pipeline()
+        def sub_pipeline():
+            node = component(component_in_path=component_input)
+            node.outputs.component_out_path.name = 'sub_pipeline_output'
+            node.outputs.component_out_path.version = 'v1'
+            return {
+                'sub_pipeine_a_output': node.outputs.component_out_path
+            }
+
+        @dsl.pipeline()
+        def register_both_output():
+            # register NodeOutput which is binding to PipelineOutput
+            node = component(component_in_path=component_input)
+            node.outputs.component_out_path.name = 'n1_output'
+            node.outputs.component_out_path.version = 'v1'
+
+            # register NodeOutput which isn't binding to PipelineOutput
+            node_2 = component(component_in_path=component_input)
+            node_2.outputs.component_out_path.name = 'n2_output'
+            node_2.outputs.component_out_path.version = 'v1'
+
+            # register NodeOutput of subgraph
+            sub_node = sub_pipeline()
+            sub_node.outputs.sub_pipeine_a_output.name = 'sub_pipeline'
```

Comment:
What's the behavior when `sub_node` don't have this setting?

Reply:
https://ml.azure.com/runs/sleepy_squash_2d1cftj8hx?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourceGroups/sdk/providers/Microsoft.MachineLearningServices/workspaces/sdk-master&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#  this pipeline is used to test set output name for subgraph.node, but it seems that it will take a long time to finish running pipeline in this ws. I will rerun this pipeline in next pr and supplement this test.